versioned_docs/version-2021fa/assignment2.md

```diff
@@ -0,0 +1,126 @@
+---
+id: assignment2
+title: Assignment 2
+---
+
+For the second assignment, you will be working with Firebase to perform CRUD
+operations and use queries on a database. In this assignment, we're going to
+create a database of songs.
+
+:::note
+While writing this assignment, please have the app run on port `8080`.
+This will make grading easier for us.
+:::
+
+## Part 1: Creating the database
+
+First, navigate to [Firebase](https://firebase.google.com/) and create a
+new project. Go to the settings icon in the upper left, and click on it. Then,
+click project settings and go to service accounts. Copy and paste the code
+snippet provided into your `index.ts` file. Generate a new private key
+and and rename it to `service-account.json`. Put this file in the root of your
+project. Go to the database section (in the navigation bar on the left) and
+create a new Firestore database.
+
+## Part 2: Manually make a collection
+
+After the Firestore database is created, make a new collection in the database
+called "songs". Within this collection, create a document called "song1".
+This document will have three fields, "name", "artist", and "rating", with
+respective values "Never Gonna Give You Up", "Rick Astley", and your
+personal rating of the song out of 5. At this point, your database should look
+something like this:
+
+<img src="/img/a2_db.png" height="120px" width="600px" />
+
+## Part 3: Create new songs
+
+Write a `POST` request with route `/createSong`. This request will simply create
+a new song document in the songs collection of the database. The id of the
+document should be AUTOMATICALLY generated, not a name of your choice.
+
+We will be passing information in the request body (`req.body`) in the following format:
+
+```json
+{
+  name: string,
```

Comment:
the quotation marks on property names is inconsistent :/ maybe just omit unless prettier is yelling at you?